src/lib/localization-hoc.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+
+import ConnectedIntlProvider from './connected-intl-provider.jsx';
+
+/*
+ * Higher Order Component to provide localiztion state. Creates a nested IntlProvider
+ * to handle Gui intl context. The component accepts an onSetLanguage callback that is
+ * called when the locale chagnes.
+ * @param {React.Component} WrappedComponent - component to provide state for
+ * @returns {React.Component} component with intl state provided from redux
+ */
+const LocalizationHOC = function (WrappedComponent) {
+    class LocalizationWrapper extends React.Component {
+        componentDidUpdate (prevProps) {
+            if (prevProps.locale !== this.props.locale) {
+                this.props.onSetLanguage(this.props.locale);
+            }
+        }
+        render () {
+            const {
+                locale, // eslint-disable-line no-unused-vars
+                onSetLanguage, // eslint-disable-line no-unused-vars
+                ...componentProps
+            } = this.props;
+            return (
+                <ConnectedIntlProvider>
+                    <WrappedComponent {...componentProps} />
+                </ConnectedIntlProvider>
+            );
+        }
+    }
+    LocalizationWrapper.propTypes = {
+        locale: PropTypes.string,
+        onSetLanguage: PropTypes.func
+    };
+
+    LocalizationWrapper.defaultProps = {
+        onSetLanguage: () => {}
```

Comment:
If someone makes it possible to call this default, they probably don't mean to (they've implemented language switching without persisting the locale anywhere). Maybe we should have the default log a warning to the console?

Reply:
No, locale switching is still in redux. If Gui is embedded in something that doesn't need it's own handling of locale (e.g. playground), then they don't need to pass in a function.

Reply:
Here's how I understand the way we intended it to work:

We want GUI to maintain its internal notion of `locale` in Redux, but we expect `locale` to come in as a prop from the outside. When we switch locale in GUI, we should call `props.onSetLanguage`. The parent should then set the locale and pass a new `locale` prop to the GUI. At that point something watching the incoming `locale` prop should notice the change and update GUI's locale redux state (maybe that would be this wrapper?).

If all of that were true, then we wouldn't want GUI to set the Redux state except if the incoming `locale` prop were to change, so that's why I'd think we'd want to warn in case of an undefined `onSetLocale`.